src/librustc/mir/visit.rs

```diff
@@ -676,20 +677,19 @@ macro_rules! make_mir_visitor {
                             place: & $($mutability)? Place<'tcx>,
                             context: PlaceContext,
                             location: Location) {
-                match place {
-                    Place::Base(place_base) => {
-                        self.visit_place_base(place_base, context, location);
-                    }
-                    Place::Projection(proj) => {
-                        let context = if context.is_mutating_use() {
-                            PlaceContext::MutatingUse(MutatingUseContext::Projection)
-                        } else {
-                            PlaceContext::NonMutatingUse(NonMutatingUseContext::Projection)
-                        };
+                let mut context = context;
 
-                        self.visit_projection(proj, context, location);
-                    }
+                if let Some(box proj) = & $($mutability)? place.projection {
+                    context = if context.is_mutating_use() {
+                        PlaceContext::MutatingUse(MutatingUseContext::Projection)
+                    } else {
+                        PlaceContext::NonMutatingUse(NonMutatingUseContext::Projection)
+                    };
+
+                    self.visit_projection(& $($mutability)? place.base, proj, context, location);
                 }
+
+                self.visit_place_base(& $($mutability)? place.base, context, location);
```

Comment:
Is it intentional that the `place_base` is visited _after_ the projections with the updated context?

Reply:
@mark-i-m yeah, this was due to a rebase problem I think. I've fixed this yesterday but thanks for checking :)

Reply:
@mark-i-m yes, this is exactly what the previous code did if you follow the recursion.